students/mayc4t/lesson08/Circle.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python
+# #############################################################################
+# Written By: Mayc4t
+# May 28, 2018
+# Self-paced Python
+# lesson 8 - Properties and Magic Method
+# fileID: Circle.py
+# #############################################################################
+
+
+
+import math
+
+class Circle:
+    def __init__(self, r):
+        self._r = r
+
+    @classmethod
+    def from_diameter(self, d):
+        #return cls(d/2)
+        return self(d/2)
```

Comment:
instead of self it should be `cls`, the name technically doesn't matter but using anything else is an anti-pattern.

Reply:
I'm not sure how... but the reason I commented out the cls was I got compile error, and I didn't know to fix that so I changed it to self. I will try it again